cmd/koda/main.go

```diff
@@ -0,0 +1,63 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"github.com/mindtastic/koda"
+	"github.com/mindtastic/koda/store/localfile"
+	"net/http"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/mindtastic/koda/log"
+)
+
+var addr = flag.String("addr", ":8000", "Address to listen on for API connections")
+var dbpath = flag.String("db", "/data/db/koda.db", "File to store database")
+
+type application struct {
+	mu         sync.RWMutex
+	store      koda.Store
+	httpServer *http.Server
+}
+
+func main() {
+	flag.Parse()
+
+	lfs := localfile.New()
+	if err := lfs.InitializePersistence(*dbpath); err != nil {
+		log.Fatalf("error initializing database: %v", err)
+	}
+
+	app := &application{
+		httpServer: &http.Server{
+			Addr: *addr,
+		},
+		store: lfs,
+	}
+
+	app.httpServer.Handler = app.initializeMux()
+	shutdown := make(chan os.Signal)
+	signal.Notify(shutdown, os.Interrupt, syscall.SIGTERM)
+
+	go func() {
+		if err := app.httpServer.ListenAndServe(); err != nil && err != http.ErrServerClosed {
+			log.Fatalf("error listening on address %q: %v", app.httpServer.Addr, err)
```

Comment:
If the binary exits, the pod will be restarted. It can _also_ be restarted due to bad response on healthchecks but the current implementation does not cover that yet.